android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt

```diff
@@ -85,11 +85,18 @@ class GestureHandlerOrchestrator(
   private fun hasOtherHandlerToWaitFor(handler: GestureHandler<*>) =
     gestureHandlers.any { !isFinished(it.state) && shouldHandlerWaitForOther(handler, it) }
 
-  private fun shouldBeCancelledByFinishedHandler(handler: GestureHandler<*>) = gestureHandlers.any { shouldHandlerWaitForOther(handler, it) && it.state == GestureHandler.STATE_END }
+  private fun shouldBeCancelledByFinishedHandler(handler: GestureHandler<*>) =
+    gestureHandlers.any { shouldHandlerWaitForOther(handler, it) && it.state == GestureHandler.STATE_END }
+
+  private fun handlersWithinHandlerBounds(handler: GestureHandler<*>) =
+    gestureHandlers.filter { it.isWithinBounds(wrapperView, handler.x, handler.y) }
+
+  private fun shouldBeCancelledByActiveHandler(handler: GestureHandler<*>) =
+    handlersWithinHandlerBounds(handler).any { isActive(it.state) && !canRunSimultaneously(handler, it) }
```

Comment:
Can't we use `hasCommonPointers` instead?

Reply:
you're right, changed to `hasCommonPointers` in [ba8596c](https://github.com/software-mansion/react-native-gesture-handler/pull/3095/commits/ba8596c7a503617e9504db4dd746aa342f544a0c)